unbit/uwsgi
# Fix for request URLs containing spaces

This patch fixes bug with spaces in request url.

How to reproduce bug:

* Use following WSGI app:
```
def application(env, start_response):
    start_response('200 OK', [('Content-Type','text/html')])
    return ["REQUEST_URI: {REQUEST_URI}\nPATH_INFO: {PATH_INFO}\n".format(**env)]
```
* Make following request (with literal spaces in URL)
`curl -v 'http://127.0.0.1:5555/h ello?a=o my'`

Expected result: server conforms HTTP protocol
Actual result: server returns malformed HTTP response:
```
[nuf@dsmal-work ~]$ curl -v 'http://127.0.0.1:5555/h ello?a=o my '
*   Trying 127.0.0.1...
* Connected to 127.0.0.1 (127.0.0.1) port 5555 (#0)
> GET /h ello?a=o my HTTP/1.1
> Host: 127.0.0.1:5555
> User-Agent: curl/7.47.0
> Accept: */*
> 
ello?a=o my god HTTP/1.1 200 OK
Content-Type: text/html

REQUEST_URI: /h
PATH_INFO: /h
* Connection #0 to host 127.0.0.1 left intact
```
